third_party/amd/backend/compiler.py

```diff
@@ -204,20 +204,25 @@ def make_ttgir(mod, metadata, options):
         amd.passes.ttgpuir.add_optimize_epilogue(pm)
         passes.ttgpuir.add_optimize_dot_operands(pm, True)
 
-        stream_prefetch = os.getenv("TRITON_HIP_STREAM_PREFETCH", "0") == "1"
         use_buffer_ops = os.environ.get("AMDGCN_USE_BUFFER_OPS", "0") == "1"
 
         # The `local-prefetch` scheduling variant requires turning on buffer ops.
         if options.instruction_sched_variant == "local-prefetch":
-            stream_prefetch = True
+            use_buffer_ops = True
 
         if amd.has_matrix_core_feature(options.arch):
             assert options.num_stages != 0, ("Triton AMD backend pipeliner has been updated. "
                                              "We used to trigger software pipelining with "
                                              "num_stages == 0. Now it will not happen anymore; "
                                              "please update to use num_stages == 2 for "
                                              "equivalent behavior in the past.")
-            amd.passes.ttgpuir.add_stream_pipeline(pm, options.num_stages, stream_prefetch)
+            stream_stages = os.getenv("TRITON_HIP_STREAM_STAGES", "0,0,-1,-1").split(',')
+            if len(stream_stages) != 4:
+                ## print warning message
+                stream_stages = [0, 0, -1, -1]
+            if options.instruction_sched_variant == "local-prefetch":
+                stream_stages = [0, 1, -2, -1]
```

Comment:
What is the assumption for local_prefetch? Does it require num_stages > 2?

Reply:
It does work for num_stages=2 but keeps GLOBAL_LOAD,LOCAL_STORE,LOCAL_LOAD in stage 0. 

We now require num_stages explicitly to set to avoid confusion. So setting num_stages=3 and instruction_sched_hint=local-prefetch will get the previous result.